charts/nginx-ingress-services/templates/ingress.yaml

```diff
@@ -4,11 +4,38 @@
 apiVersion: {{ include "ingress.apiVersion" . }}
 kind: Ingress
 metadata:
-  name: nginx-ingress
+  name: nginx-ingress-{{ .Values.ingressName }}
   annotations:
     {{- if not $ingressFieldNotAnnotation }}
     kubernetes.io/ingress.class: "{{ .Values.config.ingressClass }}"
     {{- end }}
+    {{ if .Values.config.renderCSPInIngress }}
+    {{ if ne .Values.config.ingressClass "nginx"}}
+        {{ fail "In ingress CSP header setting only works with 'nginx' controller." }}
```

Comment:
If this were changed in CI integration tests, we'd run into a problem, as in 

```
hack/helm_vars/nginx-ingress-controller/values.yaml.gotmpl
6:    ingressClass: "nginx-{{ .Release.Namespace }}"
```

We set the ingress class to something different matching the ingress controller. Can the condition be changed to be "if "nginx" is a substring of ....Values.ingressClass" (however the syntax for that is in gotemplate, so as not to conflict here, were we to test this in integration tests?

Reply:
I've added a check if the `ingressClass` contains `"nginx"`. Feels a bit hand-wavy, but I don't have a better idea / better field to check.